src/main/java/me/arminb/sara/dao/UserDAOImpl.java

```diff
@@ -99,14 +95,11 @@ public User findById(ObjectId id) throws DataAccessException {
             if (email != null) {
```

Comment:
change the query to use regex instead of equality. You can do it like this:
`query.append("username", new BasicDBObject("$regex", ".*" + username + ".*"));`